servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SniTest.java

```diff
@@ -26,109 +25,128 @@
             assertThrows(SSLHandshakeException.class, () -> client.request(client.get("/")));
         }
     }
 
-    @ParameterizedTest(name = "protocols={0}, alpn={1}")
-    @MethodSource("protocolsAndAlpn")
-    void sniClientDefaultServerSuccess(List<HttpProtocol> protocols, boolean useALPN) throws Exception {
-        try (ServerContext serverContext = HttpServers.forAddress(localAddress(0))
-                .protocols(protocolConfigs(protocols))
-                .sslConfig(trustedServerConfig(alpnIds(protocols, useALPN)))
+    @ParameterizedTest(name = "serverSslProvider={0} clientSslProvider={1} protocol={2} useALPN={3}")
+    @MethodSource("params")
+    void sniClientDefaultServerSuccess(SslProvider serverSslProvider, SslProvider clientSslProvider,
+                                       HttpProtocol protocol, boolean useALPN) throws Exception {
+        try (ServerContext serverContext = newServerBuilder(SERVER_CTX, protocol)
+                .sslConfig(trustedServerConfig(serverSslProvider, alpnIds(protocol, useALPN)))
                 .listenBlockingAndAwait(newSslVerifyService());
-             BlockingHttpClient client = newClient(serverContext, protocols, useALPN)) {
+             BlockingHttpClient client = newClient(serverContext, clientSslProvider, protocol, useALPN)) {
             assertEquals(HttpResponseStatus.OK, client.request(client.get("/")).status());
         }
     }
 
-    @ParameterizedTest(name = "protocols={0}, alpn={1}")
-    @MethodSource("protocolsAndAlpn")
-    void noSniClientDefaultServerFallbackSuccess(List<HttpProtocol> protocols, boolean useALPN) throws Exception {
-        try (ServerContext serverContext = HttpServers.forAddress(localAddress(0))
-                .protocols(protocolConfigs(protocols))
-                .sslConfig(trustedServerConfig(alpnIds(protocols, useALPN)),
-                        singletonMap("no_match" + SNI_HOSTNAME, untrustedServerConfig()))
+    @ParameterizedTest(name = "serverSslProvider={0} clientSslProvider={1} protocol={2} useALPN={3}")
+    @MethodSource("params")
+    void noSniClientDefaultServerFallbackSuccess(SslProvider serverSslProvider, SslProvider clientSslProvider,
+                                                 HttpProtocol protocol, boolean useALPN) throws Exception {
+        try (ServerContext serverContext = newServerBuilder(SERVER_CTX, protocol)
+                .sslConfig(trustedServerConfig(serverSslProvider, alpnIds(protocol, useALPN)),
+                        singletonMap("no_match" + SNI_HOSTNAME, untrustedServerConfig(serverSslProvider)))
                 .listenBlockingAndAwait(newSslVerifyService());
              BlockingHttpClient client = HttpClients.forSingleAddress(
                      getLoopbackAddress().getHostName(),
                      serverHostAndPort(serverContext).port())
-                     .protocols(protocolConfigs(protocols))
+                     .protocols(protocol.config)
```

Comment:
I see that this PR changes from a list of protocols to a single protocol (everywhere?). For my own curiosity, are we loosing any test coverage now by providing only one vs. multiple?

Reply:
No, the previous code always used `singletonList`